hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyRequest.java

```diff
@@ -536,6 +538,40 @@ protected void checkBucketQuotaInBytes(OmBucketInfo omBucketInfo,
             + allocateSize) + " Bytes.",
             OMException.ResultCodes.QUOTA_EXCEEDED);
       }
+    } else {
+      checkVolumeQuotaInBytes(metadataManager, omBucketInfo.getVolumeName(),
+          allocateSize);
+    }
+  }
+
+  private void checkVolumeQuotaInBytes(
+      OMMetadataManager metadataManager, String volumeName,
+      long allocateSize) throws IOException {
+    String volumeKey = metadataManager.getVolumeKey(volumeName);
+    OmVolumeArgs omVolumeArgs = metadataManager.getVolumeTable()
+        .get(volumeKey);
+    if (omVolumeArgs.getQuotaInBytes() <= OzoneConsts.QUOTA_RESET) {
+      return;
+    }
+
+    long totalBucketQuota = 0L;
+    List<OmBucketInfo> bucketList = metadataManager.listBuckets(
+        volumeName, null, null, Integer.MAX_VALUE);
+    for (OmBucketInfo bucketInfo : bucketList) {
+      long nextQuotaInBytes = bucketInfo.getQuotaInBytes();
+      if (nextQuotaInBytes > OzoneConsts.QUOTA_RESET) {
+        totalBucketQuota += nextQuotaInBytes;
```

Comment:
If quota is enabled on the bucket then, (nextQuotaInBytes > OzoneConsts.QUOTA_RESET) will return true. I think even in here we need to add up the usedBytes of bucket instead of quota.

Reply:
@sadanand48 
Here, bucket quota is exclusive, if quota is assigned to bucket, then its reserved by the bucket. So that is considered, not used byte.
Otherwise, if total quota may cross in case
bucket with quota -- used one is less
other bucket object is created considering only use one
later bucket with quota is occupied (considering reserved quota)
Then total bucket used quota may exceed

Reply:
Lets say there is a single volume vol1 and 2 buckets with quota enabled buck1 and buck2, The quota on volume set is 2GB and each bucket has a quota set of 1GB , In this case , the totalBuckeQuota will be 2GB and  we try to write a non-zero size key `(omVolumeArgs.getQuotaInBytes() - totalBucketQuota < allocateSize) `will return true throwing exception even though the space is not fully utilized?

Reply:
This is applicable only for bucket where quota is not set. So above scenario has no impact.